service-front/module/Application/src/Controller/DonorFlowController.php

```diff
@@ -186,14 +186,19 @@ public function drivingLicenceNumberAction(): ViewModel
         $view->setVariable('form', $form);
 
         if (count($this->getRequest()->getPost())) {
-            return $this->formProcessorService->processDrivingLicenceForm(
+            $formProcessorResponseDto = $this->formProcessorHellper->processDrivingLicenceForm(
+                $uuid,
                 $this->getRequest()->getPost(),
                 $form,
-                $view,
                 $templates
             );
-        }
 
+            foreach ($formProcessorResponseDto->getVariables() as $key => $variable) {
+                $view->setVariable($key, $variable);
+            }
```

Comment:
Looks like there are a few more loops that could benefit from the `setVariables($dto->getVariables())` treatment